capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -0,0 +1,288 @@
+
+        Map<String, Object> map;
+        for (Entity entity : results) {
+            map = new HashMap<>();
+            map.put("week", entity.getProperty("week").toString());
+            map.put("count", Integer.parseInt(entity.getProperty("count").toString()));
+            resultList.add(map);
+        }
+
+        return resultList;
+    }
+
+    /**
+     * Gets all pageViews for the entire system.
+     * @return returns a Map<String, Object> in the following format:
+     *          {
+     *              "firstDate":
+     *                  {
+     *                      "week": <int>
+     *                      "year": <int>
+     *                  },
+     *              "clickData":
+     *                  [
+     *                      {
+     *                          "restaurantName": <String>
+     *                          "week": <int>
+     *                          "year": <int>
+     *                          "numClicks": <int>
+     *                      },
+     *                      ...
+     *                  ]
+     *          }
+     *
+     *          "clickData" represents all the click data that we have.
+     *          "firstDate" represent the first time data was collected in our system.
+     */
+    public Map<String, Object> getAllPageViews() {
+        Query query = new Query("PageViews");
+        PreparedQuery preparedQuery = dataStore.prepare(query);
+        List<Entity> results = preparedQuery.asList(FetchOptions.Builder.withDefaults());
+
+        Map<String, Object> resultMap = new HashMap<>();
+
+        Map<String, Integer> firstPageView = new HashMap<>();
+        // Initialized with max value so any date will change the map at the beginning.
+        firstPageView.put("year", Integer.MAX_VALUE);
+        firstPageView.put("week", Integer.MAX_VALUE);
+
+        List<Map<String, Object>> clickData = populateClickData(results, firstPageView);
+
+        resultMap.put("firstDate", firstPageView);
+        resultMap.put("clickData", clickData);
+
+        return resultMap;
+    }
+
+    /**
+     * Parses the array of entities and populates a list of maps that contain the required information.
+     * @param entityList list of entities that we need to be turned into a map: List<Entity>
+     * @param firstPageView map of the earliest date in one of the entities: Map<String, Integer>
+     * @return returns a list of maps representative of the click data:
+     *              "clickData":
+     *                  [
+     *                      {
+     *                          "restaurantName": <String>
+     *                          "week": <int>
+     *                          "year": <int>
+     *                          "numClicks": <int>
+     *                      },
+     *                      ...
+     *                  ]
+     */
+    protected List<Map<String, Object>> populateClickData(List<Entity> entityList, Map<String, Integer> firstPageView) {
```

Comment:
Hmmm, why is this protected and not just private? Note that if the answer will be "you're using it in your test to populate some data", that's a sign that something might need to be refactored. :)

Reply:
That most definitely not the reason :P haha. It has been taken out due to the refactoring with the classes.